fipy/solvers/petsc/linearLUSolver.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import division
+from builtins import range
+from past.utils import old_div
+__docformat__ = 'restructuredtext'
+
+import os
+
+from petsc4py import PETSc
+
+from fipy.solvers.petsc.petscSolver import PETScSolver
+
+__all__ = ["LinearLUSolver"]
+
+class LinearLUSolver(PETScSolver):
+
+    """
+    The `LinearLUSolver` is an interface to the LU preconditioner in PETSc.
+    A direct solve is performed.
+
+    """
+      
+    def __init__(self, tolerance=1e-10, iterations=10, precon="lu"):
+        """
+        :Parameters:
+          - `tolerance`: The required error tolerance.
+          - `iterations`: The maximum number of iterative steps to perform.
+          - `precon`: *Ignored*. 
+
+        """
+        PETScSolver.__init__(self, tolerance=tolerance,
+                             iterations=iterations, precon="lu")
+
+    def _solve_(self, L, x, b):
+        ksp = PETSc.KSP()
+        ksp.create(PETSc.COMM_WORLD)
+        ksp.setType("preonly")
+        ksp.getPC().setType(self.preconditioner)
+        # TODO: SuperLU invoked with PCFactorSetMatSolverPackage(pc, MATSOLVERSUPERLU)
```

Comment:
Is this a todo-TODO, or an in the fullness of time TODO?

Reply:
That's an "I wrote that note [7 years ago](https://github.com/guyer/fipy/blame/ticket644-Add_support_for_PETSc_solvers/fipy/solvers/petsc/linearLUSolver.py#L38)-TODO"